code/game/objects/items.dm

```diff
@@ -411,6 +411,12 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 	return M.can_equip(src, slot, disable_warning, bypass_equip_delay_self)
 
+/obj/item/proc/mob_can_unequip(mob/living/M, mob/living/unequipper)

```

Comment:
This proc is a notable change, it allows the gun to decide whether you can unequip it or not.

If this PR doesn't get merged, i'll take this part out and PR it separately, it can be used for things like the tin foil hat.